### DISCUSSION TITLE
How to contribute and run server locally?

### DISCUSSION THREAD

**Discussion by tasnim0tantawi** [firefly-iii/firefly-iii#9148] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have tried to run server locally to contribute. I keep getting a Vite related error.
The Docs mention that the project is not updatable from github repo. Contributing page is missing.
What to do? I want to contribute.

![20240818_210641](https://github.com/user-attachments/assets/04b1b995-90fb-41c9-bf0e-a9eb1797e310)

### Debug information

The project runs fine when I download zip file in docs.

### Expected behaviour

To be able to clone project normally.

### Steps to reproduce

1. Clone the project
2. Try to run it following docs steps.
https://docs.firefly-iii.org/how-to/firefly-iii/installation/self-managed/
3. An error related to JS files will appear.

### Additional info
I know that it runs fine if I download the .zip file or I run Docker image, but I want to clone it to contribute. Please help.

**Comment by JC5** [comment]
Hey!

great to hear that you want to contribute. What do you consider your knowledge base in regards to npm and vite?

**Reply by tasnim0tantawi** [reply]
I have experience with Vite and React (still figuring out Vue and project structure).
I'm not sure, should I add vite manifest and what's missing manually? 
Thanks so much for your respond ! @JC5

**Reply by JC5** [reply, marked as answer]
The `package.json` in the root of the Firefly III directory refers to two workspaces. You can use either `--workspace=v1` or `--workspace=v2` to issue commands. Since you mention Vite, I expect you need `--workspace=v2`. Vite is only used in the new layout.

**Reply by tasnim0tantawi** [reply]
I'm not sure, but I always get the error mentioned in this discussion https://github.com/orgs/firefly-iii/discussions/8694 and the answer you suggested is following the docs and downloading .zip file, that's why I got very confused. @JC5

**Reply by tasnim0tantawi** [reply]
I will try running it now
